scripts/preprocess_spec.py

```diff
@@ -30,24 +39,67 @@ def _to_camel_case(s):
     return ''.join(_title(y) for y in s.split("_"))
 
 
-def iterate_through_operations(spec, func):
+def apply_funcs_to_spec(spec, func, *params):
+    """Apply fun to each operation in the spec.
+
+    First two parameters of func will be operation and it's parent, and the
+    rest of parameters would be anything passed after func to this function.
+    """
     for k, v in spec['paths'].iteritems():
         for op in _ops:
-            if op in v:
-                func(v[op])
+            if op not in v:
+                continue
+            if func(v[op], v, *params):
+                del v[op]
+
+
+def _has_property(prop_list, property_name):
+    for prop in prop_list:
+        if prop["name"] == property_name:
+            return True
+
+
+def remove_watch_operations(op, parent, operation_ids):
```

Comment:
(No action required)

If/when this method is targeted by a unit test, it may make sense to return messages (e.g. ``Cannot find...``) as exceptions and allow the calling function more flexibility in how those cases are handled (i.e. script would output to stdout, test would check that the expected exception was returned for a given input).